### PR TITLE
rachel - L105.nh3_an_USA_S_T_Y

### PR DESCRIPTION
**Pull request by rhoesly**
L105.nh3_an_USA_S_T_Y - Amends/adds chunk, output test file, constants, and automatically
generated files.

### CODE REVIEW

File: R/constants.R [hunk 1]
```diff
@@ -18,6 +18,8 @@ FLAG_SUM_TEST   <- "FLAG_SUM_TEST"
 # Time constants
 HISTORICAL_YEARS <- 1971:2010
 FUTURE_YEARS <- seq( 2015, 2100, 5 )
+NH3_HISTORICAL_YEARS <- 1990:2002
+NH3_EXTRA_YEARS <- 1971:1989
```
Comment (bpbond) [inline R/constants.R]: Would it be clearer/useful/accurate to have these be e.g. `emissions.NH3_HISTORICAL_YEARS`? Not sure; consider.

File: R/zchunk_L105.nh3_an_USA_S_T_Y.R [hunk 2]
```diff
@@ -53,19 +55,63 @@ module_emissions_L105.nh3_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+      dplyr::rename(emissions = value)
+
+    # Process FAO production data:
+    # Subset US data, aggregate to EPA sectors (all animal production)
+
+    L105.out_Mt_USA_an_C_Sys_Fd_Yh <- L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      filter(GCAM_region_ID == 1) %>%
+      group_by(year) %>%
+      summarize_at(vars(value), funs(sum ) ) %>%
+      dplyr::rename(production = value) %>%
```
Comment (bpbond) [inline R/zchunk_L105.nh3_an_USA_S_T_Y.R]: Could also add `rename` to line 16 above, and remove the namespace qualifier here and below.

File: R/zchunk_L105.nh3_an_USA_S_T_Y.R [hunk 3]
```diff
@@ -53,19 +55,63 @@ module_emissions_L105.nh3_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
 
+    # Process EPA NH3 emissions estimates:
+    # Map to GCAM sector and Technology, convert to Tg and aggregate
+    # Notes:
+    # - when mapping from EPA sectors to GCAM sectors and technologies there should be NAs
+    #   (don't map very aggregate or very detailed EPA sectors to avoid double counting)
+
+    L105.nh3_tg_USA_an_Sepa_F_Yh <- EPA_NH3 %>%
+      left_join(EPA_tech, by = c("Source_Category" = "EPA_Category")) %>%
+      filter(sector == 'Animals',!is.na(fuel)) %>%
```
Comment (bpbond) [inline R/zchunk_L105.nh3_an_USA_S_T_Y.R]: Why is this happening? Clearer/more comments that explain steps would help.

File: R/zchunk_L105.nh3_an_USA_S_T_Y.R [hunk 4]
```diff
@@ -53,19 +55,63 @@ module_emissions_L105.nh3_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate(year = as.numeric(year)) %>%
+      filter(year %in% NH3_HISTORICAL_YEARS) %>%
+      mutate(value = value * CONV_TST_TG) %>%  # convert from thousand short tons to Tg
+      dplyr::rename(emissions = value)
+
+    # Process FAO production data:
+    # Subset US data, aggregate to EPA sectors (all animal production)
+
+    L105.out_Mt_USA_an_C_Sys_Fd_Yh <- L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      filter(GCAM_region_ID == 1) %>%
```
Comment (bpbond) [inline R/zchunk_L105.nh3_an_USA_S_T_Y.R]: There's a global constant defined for this--use it please. `GCAM_REGION_USA` or something like that.

File: R/zchunk_L105.nh3_an_USA_S_T_Y.R [hunk 5]
```diff
@@ -53,19 +55,63 @@ module_emissions_L105.nh3_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    L105.nh3_tgmt_USA_an_Yh %>%
+      add_title("NH3 emissions factors for animals by sector / technology / year estiamted from EPA NH3 emissions and FAO production statistics") %>%
+      add_units("Tg/Mt") %>%
+      add_comments("NH3 animal emission factors for all historical years are estimates from US values") %>%
+      add_comments("Historical years 1971 - 1989 are estimated as 1990 values") %>%
       add_legacy_name("L105.nh3_tgmt_USA_an_Yh") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
+      add_precursors("common/iso_GCAM_regID",
```
Comment (bpbond) [inline R/zchunk_L105.nh3_an_USA_S_T_Y.R]: I don't see this used anywhere above? If it's not, please remove here (and from inputs as well).

**Comment by bpbond** [comment]
Nice job, Rachel. Thanks. Is this waiting on @ssmithClimate review before merging?

**Comment by rhoesly** [comment]
No, not waiting on Steve, I just tagged him to keep him informed.

**Comment by bpbond** [comment]
See also #418

**Pull request merged**